matchms/filtering/metadata_processing/derive_annotation_from_compound_name.py

```diff
@@ -55,10 +55,15 @@ def derive_smiles_from_pubchem_compound_name_search(spectrum_in: Spectrum,
             if within_mass_tolerance.shape[0] > 0:
                 # Select the match with the most
```

Comment:
Incomplete comment.

Reply:
This made me realize that I actually did the wrong thing in the line below...

Before I selected the lowest parent mass, instead of the smallest mass difference. In reality this would never give an issue, since we only select cases that are within the smallest mass difference. But it has now been fixed and a test has been added.